speechbrain/lobes/models/huggingface.py

```diff
@@ -0,0 +1,361 @@
+"""This lobe is the interface for huggingface transformers models
+It enables loading config and model via AutoConfig & AutoModel.
+
+Transformer from HuggingFace needs to be installed:
+https://huggingface.co/transformers/installation.html
+
+Authors
+ * Titouan Parcollet 2021, 2022, 2023
+ * Mirco Ravanelli 2021
+ * Boumadane Abdelmoumene 2021
+ * Ju-Chieh Chou 2021
+ * Artem Ploujnikov 2021, 2022
+ * Abdel Heba 2021
+ * Aku Rouhe 2022
+ * Arseniy Gorin 2022
+ * Ali Safaya 2022
+ * Benoit Wang 2022
+ * Adel Moumen 2022, 2023
+ * Andreas Nautsch 2022, 2023
+ * Luca Della Libera 2022
+ * Heitor Guimarães 2022
+ * Ha Nguyen 2023
+"""
+import os
+import torch
+import logging
+import pathlib
+from torch import nn
+from typing import Union
+from huggingface_hub import model_info
+from speechbrain.pretrained.fetching import fetch
+from speechbrain.dataio.dataio import length_to_mask
+
+
+# We check if transformers is installed.
+try:
+    from transformers import (
+        AutoConfig,
+        AutoTokenizer,
+        AutoFeatureExtractor,
+        AutoModelForPreTraining,
+        # AutoProcessor,
+        AutoModel,
+    )
+
+except ImportError:
+    MSG = "Please install transformers from HuggingFace to use wav2vec2 / Hubert\n"
+    MSG += "E.G. run: pip install transformers"
+    raise ImportError(MSG)
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceTransformer(nn.Module):
+    """This lobe provides AutoClass architecture loading into SpeechBrain modules.
```

Comment:
Maybe slightly more precise about what it means: being able to load any model from HuggingFace transformer.

Reply:
It would be good to give information about how to use this class as well. Like it gives an ensemble of functions that the user must override by inheriting from this class to support any HuggingFace model.